pkg/client/transport/config.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transport
+
+import "net/http"
+
+// Config holds various options for establishing a transport.
+type Config struct {
```

Comment:
The transport config is still something every consumer will have to build, right? It's nice that the client gets a `*http.Client` with a transport already built, but the options that drive that come from somewhere... what benefit is a four-level structure like this providing?
```
  transport.Config
    transport.AuthConfig
      transport.TokenAuthConfig
        Token string
```